[2021-06-06 15:40:41,173] {taskinstance.py:851} INFO - Dependencies all met for <TaskInstance: forex_data_pipeline.spark_submit_job 2021-06-05T00:00:00+00:00 [queued]>
[2021-06-06 15:40:41,212] {taskinstance.py:851} INFO - Dependencies all met for <TaskInstance: forex_data_pipeline.spark_submit_job 2021-06-05T00:00:00+00:00 [queued]>
[2021-06-06 15:40:41,214] {taskinstance.py:1042} INFO - 
--------------------------------------------------------------------------------
[2021-06-06 15:40:41,215] {taskinstance.py:1043} INFO - Starting attempt 4 of 5
[2021-06-06 15:40:41,216] {taskinstance.py:1044} INFO - 
--------------------------------------------------------------------------------
[2021-06-06 15:40:41,234] {taskinstance.py:1063} INFO - Executing <Task(SparkSubmitOperator): spark_submit_job> on 2021-06-05T00:00:00+00:00
[2021-06-06 15:40:41,245] {standard_task_runner.py:52} INFO - Started process 1190 to run task
[2021-06-06 15:40:41,254] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'forex_data_pipeline', 'spark_submit_job', '2021-06-05T00:00:00+00:00', '--job-id', '36', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/dag_spark.py', '--cfg-path', '/tmp/tmp6scrbb6h', '--error-file', '/tmp/tmphsopiuvs']
[2021-06-06 15:40:41,256] {standard_task_runner.py:77} INFO - Job 36: Subtask spark_submit_job
[2021-06-06 15:40:41,457] {logging_mixin.py:104} INFO - Running <TaskInstance: forex_data_pipeline.spark_submit_job 2021-06-05T00:00:00+00:00 [running]> on host 5b1a5e9e3fee
[2021-06-06 15:40:41,600] {taskinstance.py:1257} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=admin@localhost.com
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=forex_data_pipeline
AIRFLOW_CTX_TASK_ID=spark_submit_job
AIRFLOW_CTX_EXECUTION_DATE=2021-06-05T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2021-06-05T00:00:00+00:00
[2021-06-06 15:40:41,619] {base.py:74} INFO - Using connection to: id: spark_conn. Host: spark://spark-master, Port: 7077, Schema: , Login: marcelo, Password: XXXXXXXX, extra: None
[2021-06-06 15:40:41,624] {spark_submit.py:364} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf master=spark://spark-master:7077 --conf spark.dynamicAllocation.enabled=false --conf spark.shuffle.service.enabled=false --num-executors 0 --total-executor-cores 1 --executor-cores 1 --executor-memory 512m --driver-memory 512m --name airflow-spark /opt/airflow/dags/test.py
[2021-06-06 15:40:44,366] {spark_submit.py:526} INFO - Warning: Ignoring non-Spark config property: master
[2021-06-06 15:40:44,569] {spark_submit.py:526} INFO - WARNING: An illegal reflective access operation has occurred
[2021-06-06 15:40:44,571] {spark_submit.py:526} INFO - WARNING: Illegal reflective access by org.apache.spark.unsafe.Platform (file:/home/airflow/.local/lib/python3.7/site-packages/pyspark/jars/spark-unsafe_2.12-3.1.2.jar) to constructor java.nio.DirectByteBuffer(long,int)
[2021-06-06 15:40:44,572] {spark_submit.py:526} INFO - WARNING: Please consider reporting this to the maintainers of org.apache.spark.unsafe.Platform
[2021-06-06 15:40:44,574] {spark_submit.py:526} INFO - WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[2021-06-06 15:40:44,575] {spark_submit.py:526} INFO - WARNING: All illegal access operations will be denied in a future release
[2021-06-06 15:40:44,640] {spark_submit.py:526} INFO - Exception in thread "main" org.apache.spark.SparkException: Number of executors must be a positive number
[2021-06-06 15:40:44,641] {spark_submit.py:526} INFO - at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:631)
[2021-06-06 15:40:44,643] {spark_submit.py:526} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:264)
[2021-06-06 15:40:44,644] {spark_submit.py:526} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateArguments(SparkSubmitArguments.scala:234)
[2021-06-06 15:40:44,645] {spark_submit.py:526} INFO - at org.apache.spark.deploy.SparkSubmitArguments.<init>(SparkSubmitArguments.scala:119)
[2021-06-06 15:40:44,646] {spark_submit.py:526} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2$$anon$3.<init>(SparkSubmit.scala:1022)
[2021-06-06 15:40:44,647] {spark_submit.py:526} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.parseArguments(SparkSubmit.scala:1022)
[2021-06-06 15:40:44,648] {spark_submit.py:526} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:85)
[2021-06-06 15:40:44,650] {spark_submit.py:526} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1039)
[2021-06-06 15:40:44,650] {spark_submit.py:526} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1048)
[2021-06-06 15:40:44,652] {spark_submit.py:526} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2021-06-06 15:40:44,705] {taskinstance.py:1455} ERROR - Cannot execute: spark-submit --master spark://spark-master:7077 --conf master=spark://spark-master:7077 --conf spark.dynamicAllocation.enabled=false --conf spark.shuffle.service.enabled=false --num-executors 0 --total-executor-cores 1 --executor-cores 1 --executor-memory 512m --driver-memory 512m --name airflow-spark /opt/airflow/dags/test.py. Error code is: 1.
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1112, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1285, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1315, in _execute_task
    result = task_copy.execute(context=context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 183, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 455, in submit
    self._mask_cmd(spark_submit_cmd), returncode
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf master=spark://spark-master:7077 --conf spark.dynamicAllocation.enabled=false --conf spark.shuffle.service.enabled=false --num-executors 0 --total-executor-cores 1 --executor-cores 1 --executor-memory 512m --driver-memory 512m --name airflow-spark /opt/airflow/dags/test.py. Error code is: 1.
[2021-06-06 15:40:44,722] {taskinstance.py:1503} INFO - Marking task as UP_FOR_RETRY. dag_id=forex_data_pipeline, task_id=spark_submit_job, execution_date=20210605T000000, start_date=20210606T154041, end_date=20210606T154044
[2021-06-06 15:40:44,784] {local_task_job.py:146} INFO - Task exited with return code 1
